src/pharmpy/tools/structsearch/drugmetabolite.py

```diff
@@ -1,69 +1,105 @@
+        and len(metabolite_functions) == 0
+    ):
+        raise ValueError(
+            'Require at least one metabolite model type.'
+            ' Try adding METABOLITE(BASIC) or METABOLITE(PSC) to search space'
+        )
+
+    if model.statements.ode_system.find_compartment('METABOLITE'):
+        base_description = model.description
     else:
-        presystemic_option = [False, True]
-    for presystemic in presystemic_option:
-        candidate_model = add_metabolite(model, presystemic=presystemic)
-        candidate_model = set_name(candidate_model, 'presystemic')
-
-        candidate_model = candidate_model.replace(parent_model='base_metabolite')
-        if presystemic:
-            models.append(candidate_model)
-
-        candidate_model = add_peripheral_compartment(candidate_model, "METABOLITE")
-        candidate_model = set_name(
-            candidate_model, f'{"presystemic" if presystemic else "base_metabolite"}_peripheral'
+        base_description = determine_base_description(
+            metabolite_mfl_statements, peripheral_mfl_statements
+        )
+
+    wb = WorkflowBuilder(name="drug_metabolite")
+
+    start_task = Task("start", _start, model, results)
+    wb.add_task(start_task)
+
+    def apply_transformation(eff, f, model_entry):
+        candidate_model = f(model_entry.model)
+        candidate_model = candidate_model.replace(
+            name="TEMP", description='_'.join(eff), modelfit_results=None
         )
-        if presystemic:
-            candidate_model = candidate_model.replace(parent_model='presystemic')
+        return ModelEntry.create(model=candidate_model, modelfit_results=None, parent=model)
 
-        models.append(candidate_model)
+    for eff, func in metabolite_functions.items():
+        candidate_met_task = Task(str(eff), apply_transformation, eff, func)
+        wb.add_task(candidate_met_task, predecessors=start_task)
 
-    return models
+    if len(peripheral_mfl_statements) == 0:
+        candidate_model_tasks = []
+        model_index = 1
+        for out_task in wb.output_tasks:
+            change_name_task = Task("number_run", change_name, model_index)
+            model_index += 1
+            wb.add_task(change_name_task, predecessors=[out_task])
+            wf_fit = create_fit_workflow(n=1)
+            wb.insert_workflow(wf_fit, predecessors=[change_name_task])
+
+            candidate_model_tasks += wf_fit.output_tasks
+    else:
+        wb, candidate_model_tasks = exhaustive_stepwise(
+            peripheral_functions, "no_add", wb, "structsearch"
+        )
+
+    return WorkflowBuilder(wb), candidate_model_tasks, base_description
+
+
+def _start(model, results):
+    return ModelEntry.create(model=model, modelfit_results=results)
+
+
+def change_name(index, modelentry):
+    return ModelEntry.create(
+        model=modelentry.model.replace(name=f'structsearch_run{index}'),
+        modelfit_results=modelentry.modelfit_results,
```

Comment:
Will it have results at this point? It's before fitting right?

Reply:
You're right! It's fitted just after!